cmd/fluxctl/completion_cmd.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	completionShells = map[string]func(out io.Writer, cmd *cobra.Command) error{
+		"bash": runCompletionBash,
+		"zsh":  runCompletionZsh,
+	}
+)
+
+func newCompletionCommand() *cobra.Command {
+	shells := []string{}
+	for s := range completionShells {
+		shells = append(shells, s)
+	}
+
+	return &cobra.Command{
+		Use:                   "completion shell",
+		DisableFlagsInUseLine: true,
+		Short:                 "Output shell completion code for the specified shell (bash or zsh)",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return newUsageError("please specify a shell")
+			}
+
+			if len(args) > 1 {
+				return newUsageError("please specify either bash or zsh")
```

Comment:
Making use of `shells` here would make this error maintenance free.

Reply:
Good point! 👍